Documentation improvements (#1788)
Azure/azure-service-operator: docs/v2/adding-new-codegenerated-resource.md

## Fix any errors raised by the code generator

### \<Resource\> looks like a resource reference but was not labelled as one
Example:

> >  Replace cross-resource references with genruntime.ResourceReference: ["github.com/Azure/azure-service-operator/hack/generated/_apis/microsoft.containerservice/v1alpha1api20210501/PrivateLinkResource.Id" looks like a resource reference but was not labelled as one. It might need to be manually added to `newKnownReferencesMap`,

Reviewer: Suggest adding some line breaks so the message can be read without scrolling.
```suggestion
>  Replace cross-resource references with genruntime.ResourceReference: 
> ["github.com/Azure/azure-service-operator/hack/generated/_apis/microsoft.containerservice/v1alpha1api20210501/PrivateLinkResource.Id" looks like a resource reference but was not labelled as one. 
> It might need to be manually added to `newKnownReferencesMap`,
```

Author: Done

> If the code generator fails with this error you must update the `newKnownReferencesMap` function in [add_cross_resource_refernces.go](hack/generator/pkg/codegen/pipeline/add_cross_resource_references.go#L185). In order to make this update you must determine if the property in question is an ARM ID or not.

Reviewer: Suggest putting the motivation for the change up front in this paragraph.

```suggestion
To fix this error, determine whether the property in question is an ARM ID or not, and then update the `newKnownReferencesMap()` function in [add_cross_resource_refernces.go](hack/generator/pkg/codegen/pipeline/add_cross_resource_references.go#L185).
```

Author: Improved as you suggested.

> If the code generator fails with this error you must update the `newKnownReferencesMap` function in [add_cross_resource_refernces.go](hack/generator/pkg/codegen/pipeline/add_cross_resource_references.go#L185). In order to make this update you must determine if the property in question is an ARM ID or not.

Reviewer 2: this this a typo?
```suggestion
If the code generator fails with this error you must update the `newKnownReferencesMap` function in [add_cross_resource_references.go](hack/generator/pkg/codegen/pipeline/add_cross_resource_references.go#L185). In order to make this update you must determine if the property in question is an ARM ID or not.
```

Author: yes.

If the property is an ARM ID, update `newKnownReferencesMap` to flag that property as a reference:
```go
{
       typeName: astmodel.MakeTypeName(configuration.MakeLocalPackageReference("microsoft.containerservice", "v1alpha1api20210501"), "PrivateLinkResource"),
       propName: "Id", 
}: true,
```

> If the property is not always an ARM ID, update `newKnownReferencesMap` to indicate that property is not a reference:

Reviewer: I'm looking at these two code snippets and thinking we should highlight that they're only difference is the boolean value used.

Perhaps:
```suggestion
If the property is not an ARM ID, update `newKnownReferencesMap` to indicate that property is not a reference by providing the value **false** instead:
```

Author: Done
